src/cairo_runner.rs

```diff
@@ -230,6 +237,65 @@ impl PyCairoRunner {
             .ok_or_else(|| PyTypeError::new_err(MEMORY_GET_SEGMENT_USED_SIZE_MSG))?
             .to_object(py))
     }
+
+    #[allow(clippy::too_many_arguments)]
+    pub fn run_from_entrypoint(
+        &mut self,
+        entrypoint: &PyAny,
+        args: Vec<&PyAny>,
+        typed_args: Option<bool>,
+        verify_secure: Option<bool>,
+        apply_modulo_to_args: Option<bool>,
+    ) -> PyResult<()> {
+        enum Either {
+            MaybeRelocatable(MaybeRelocatable),
+            VecMaybeRelocatable(Vec<MaybeRelocatable>),
+        }
+
+        impl Either {
+            pub fn as_any(&self) -> &dyn Any {
+                match self {
+                    Self::MaybeRelocatable(x) => x as &dyn Any,
+                    Self::VecMaybeRelocatable(x) => x as &dyn Any,
+                }
+            }
+        }
```

Comment:
I haven't tested it, but it wouldn't make sense to cast the args directly to `&PyAny` since `&PyAny as &dyn Any` shouldn't downcast to `MaybeRelocatable` or `Vec<MaybeRelocatable>` as far as I know.

A different solution would be to build a `Vec<Box<&dyn Any>>` which should work, but requires one heap allocation for `MaybeRelocatable`s or two for `Vec<MaybeRelocatable>`s. Either way, a conversion from the Python seems necessary.